daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -105,6 +105,9 @@ class VariableInstance private (
 
   lazy val defaultValueExpr = defaultValueExprArg
 
+  // This represents the default value at the start of processing, provided by
+  // either the defaultValue expression or by an external binding
+  var defaultValue: DataValuePrimitiveNullable = DataValue.NoValue
```

Comment:
I think my most recent push addresses these changes, but it still feels a little awkward copying around the firstInstanceInitialValue as a NVI doesn't start with a copy of the original variable instance.  Let me know what you think.

Reply:
I agree, the copying of the ``fisrtInstanceInitialValue`` thing does seem a bit weird. What I was trying to avoid by suggesting this was a member variable that only has a value in the first instance. If we didn't do this copy and someone accidentally accessed this member from an instance other than the first one, then they'll probably get a null value or some unexpected behavior. By copying the value from the first instance to all following instances, it doesn't matter from which instance you access this value, you will always get the initial value of the first instance.

Maybe a way to avoid that would be do have another class, e.g.
```scala
class InitialVariableInstance extends VariableInstance {
  var initialValue = ...
}
```
This way only the ``IntialVariableInstance`` has this initialValue variable, and so you can never accidentally access it from some other instance, because other instances don't even have this variable. I'm not sure if that's worth it though. You end up needing something like this to get the initial value:
```scala
instances(0).asInstanceOf[InitialVariableInstance].initialValue
```

I'm sure we could come up with additionl member variables/functions to get around this and make it look cleaner, but I'm not convinced it's entirely worth it for just one member.

Reply:
Note, it would probably be worth added a comment to this affect, explaing why we have this initial thing and why we are copying. Assuming it stays and we don't come up with a better alternative.